File: .leetcode/55.跳跃游戏.cpp
/*
 * @lc app=leetcode.cn id=55 lang=cpp
 *
 * [55] 跳跃游戏
 */

// @lc code=start
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool canJump(vector<int>& nums) {
        int cover = 0;
        if(nums.size()==1)return true;
        for(int i=0;i<=cover;i++){
            cover = max(i+nums[i],cover);
            if(cover>=nums.size()-1)return true;
        }
        return false;
    }
};
// @lc code=end

File: .leetcode/55.跳跃游戏.java
/*
 * @lc app=leetcode.cn id=55 lang=java
 *
 * [55] 跳跃游戏
 */

// @lc code=start
class Solution {
    public boolean canJump(int[] nums) {
        int dis = 0;
        int n = nums.length;
        for(int i = 0;i < n;i++){
            if(dis >= i){
                dis = Math.max(dis, i + nums[i]);
            }
        }
        return dis >= n - 1;
    }
}
// @lc code=end

